workbench/src/renderer/components/InvestTab/index.jsx

```diff
@@ -51,14 +53,29 @@ class InvestTab extends React.Component {
 
   async componentDidMount() {
     const { job } = this.props;
-    const {
-      args, ui_spec, ...modelSpec
-    } = await getSpec(job.modelRunName);
-    this.setState({
-      modelSpec: modelSpec,
-      argsSpec: args,
-      uiSpec: ui_spec,
-    }, () => { this.switchTabs('setup'); });
+    // start up the server for this model
+    const pid = await ipcRenderer.invoke(
+      ipcMainChannels.INVEST_SERVE,
+      job.modelRunName
+    );

```

Comment:
I wonder if we can streamline this a bit for the most common path, which is opening one of the core models. I don't expect this is too slow, it's just a little bit counter-intuitive because in almost every case, we're _not_ starting a new invest server here. If I have it right, the steps are,

1. IPC to call `INVEST_SERVE` -- but wait, the server is already running, right?
2. `settingsStore.get` to check if we're opening a "core" model or a "plugin"
3. if it's a "core" model, bail out of `createPythonFlaskProcess` without starting a new server

Do you think it's worth considering passing in some `prop` to `InvestTab` that indicates whether it's a core model or plugin? And then only call out to `createPythonFlaskProcess` if we really intend to create a new process?

Reply:
I agree it's not very intuitive. I have some concern about passing a prop indicating whether it's a core or plugin model... since that's static information that's associated with the model id, it feels redundant to pass _both_ the model id and plugin status. I'll try and think of a way to streamline this more.

Reply:
Ok, new approach:
- separate functions for creating the core server vs plugin servers
- replace `INVEST_SERVE` channel with `LAUNCH_PLUGIN_SERVER` channel
- only call `LAUNCH_PLUGIN_SERVER` if needed